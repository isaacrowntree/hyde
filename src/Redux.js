import { compose, createStore } from 'redux';
import { config } from './config';
import persistState from 'redux-localstorage';

export const AUTHENTICATE = 'AUTHENTICATE';
export const LOGOUT = 'LOGOUT';
export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';

export const Reducer =
    (state = {
      auth: {authenticated: false, failed: false},
      msg: {success: '', error: ''}
    }, action) =>
{
  if (action.type === AUTHENTICATE) {
    if (action.payload !== config.password) {
      return Object.assign({}, state, {authenticated: false, failed: true});
    }
    return Object.assign({}, state, {auth: {authenticated: true, failed: false}});
  }
  if (action.type === LOGOUT) {
    return Object.assign({}, state, {auth: {authenticated: false}});
  }
  if (action.type === SUCCESS) {
    return Object.assign({}, state, {msg: {success: action.payload, error: ''}});
  }
  if (action.type === ERROR) {
    return Object.assign({}, state, {msg: {success: '', error: action.payload}});
  }
  return state;
};

let enhancer = compose();

if (config.environment !== 'test') {
  enhancer = compose(
    persistState('auth'),
  );
}

export const Store = createStore(
  Reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  enhancer,
);
