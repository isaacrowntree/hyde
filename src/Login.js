import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { Store, AUTHENTICATE } from './Redux';
import { connect } from 'react-redux';

class Login extends Component {
  constructor(props) {
    super(props);

    this.login = this.login.bind(this);
  }

  login(e) {
    Store.dispatch({type: AUTHENTICATE, payload: e.target.firstChild.firstChild.value});
    e.preventDefault();
  }

  get loginWarning() {
    if (!this.props.failed) return null;

    return (
      <div className="alert alert-warning" role="alert">
        Invalid password.
      </div>
    );
  }

  render() {
    if (this.props.authenticated) {
      return (<Redirect to="/" />);
    }

    return (
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-8 col-sm-6 col-md-4">
          <br />
            <div className="jumbotron">
              <h1>Login</h1>

              {this.loginWarning}

              <form onSubmit={this.login}>
                <div className="form-group">
                  <input type="password" className="form-control" id="password" placeholder="Password"/>
                </div>
                <button type="submit" className="btn btn-primary pull-right">Submit</button>
              </form>
              <br />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Login = connect(state => state)(Login);

export default Login;
