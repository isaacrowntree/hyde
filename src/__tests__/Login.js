import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';
import { Provider } from 'react-redux';
import { Store } from '../Redux';

test('Login renders correctly', () => {
  const tree = renderer
    .create(<Provider store={Store}><Login /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
