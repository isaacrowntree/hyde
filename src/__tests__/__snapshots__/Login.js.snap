// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-8 col-sm-6 col-md-4"
    >
      <br />
      <div
        className="jumbotron"
      >
        <h1>
          Login
        </h1>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary pull-right"
            type="submit"
          >
            Submit
          </button>
        </form>
        <br />
      </div>
    </div>
  </div>
</div>
`;
